### PR TITLE
Correctly read blocks of fixed size from the pipe.

### PR DESCRIPTION
**Pull request by rasky**
Before this patch, the code was using io.Reader.Read(), which returns
whatever block of data is available on the pipe. If the client code
wrote into the pipe small chunks of data, Read() would return them
as-is. This meant that the code would send to the network blocks
smaller than the configured size, which the TFTP client would interpret
as file terminators.

By using io.ReadFull() we guarantee that we always read and fill a
full block of data, unless a EOF occurs. This fixes the transfer for
client code writing chunk of arbitrary size to the pipe.

### CODE REVIEW

**Comment by rasky** [comment]
To reproduce the bug on the previous code, just write a tftpHandler that writes a file into the pipe one byte at a time

File: sender.go [hunk 1]
```diff
@@ -31,20 +31,19 @@ func (s *sender) Run(isServerMode bool) {
...
 	var blockNumber uint16
 	blockNumber = 1
-	lastBlockSize := -1
 	for {
-		c, readError := s.reader.Read(buffer)
-		if readError != nil {
-			if readError == io.EOF { // && c == 0 ?
-				// we could have c != 0 here
-				if c != 0 {
-					panic("error!")
```
Comment (pin) [inline sender.go]: Uff, that was bad!

**Comment by pin** [comment]
Hi Giovanni,

I really like your fix and detailed comments!
Just wrote a simple unit test which is failing now on master: https://github.com/pin/tftp/pull/4

Thank you for your PR!

**Comment by pin** [comment]
Merged!

build status

**Pull request merged**